### PR TITLE
Added mapFree and Functor instance

### PR DESCRIPTION
**Pull request by cryogenian**

### CODE REVIEW

File: src/Data/Semiring/Free.purs [hunk 1]
```diff
@@ -30,6 +39,12 @@ liftFree f (Free xss) = sum (map (product <<< map f) xss)
 -- | `Free` is left adjoint to the forgetful functor from `Semiring`s to types.
 lowerFree :: forall a s. (Semiring s) => (Free a -> s) -> a -> s
 lowerFree f a = f (free a)
+
+sequenceFree :: forall a m. (Monad m) => Free (m a) -> m (Free a)
+sequenceFree freeM = do
+  bss <- sequence $ sequence <$> (runFree freeM)
+  let freess = (free <$>) <$> bss
```
Comment (paf31) [inline src/Data/Semiring/Free.purs]: Why not just use the constructor?

File: src/Data/Semiring/Free.purs [hunk 2]
```diff
@@ -30,6 +39,12 @@ liftFree f (Free xss) = sum (map (product <<< map f) xss)
 -- | `Free` is left adjoint to the forgetful functor from `Semiring`s to types.
 lowerFree :: forall a s. (Semiring s) => (Free a -> s) -> a -> s
 lowerFree f a = f (free a)
+
+sequenceFree :: forall a m. (Monad m) => Free (m a) -> m (Free a)
```
Comment (paf31) [inline src/Data/Semiring/Free.purs]: Could you please make this into a `Traversable` instance?

File: src/Data/Semiring/Free.purs [hunk 3]
```diff
@@ -19,7 +23,12 @@ runFree (Free xs) = xs
 -- | Lift a value of type `a` to a value of type `Free a`
 free :: forall a. a -> Free a
 free a = Free [[a]]
- 
+
+-- | Apply function `fn` to every element of `Free a` 
+mapFree :: forall a b. (a -> b) -> Free a -> Free b
```
Comment (paf31) [inline src/Data/Semiring/Free.purs]: Looks good, but I think we should prefer the `Functor` instance. Could you please just move this inside there?

**Comment by cryogenian** [comment]
Added `Foldable`, `Traversable`, `Applicative`. Can't figure out how to implement `Bind`. Will try to do it later.

**Comment by paf31** [comment]
Thanks!

**Pull request merged**